src/Console/Shell.php

```diff
@@ -472,12 +472,64 @@ public function runCommand($argv, $autoMethod = false, $extra = [])
             return $this->main(...$this->args);
         }
 
-        $this->out($this->OptionParser->help($command));
+        $this->outputUnknownCommand($command);
 
         return false;
     }
 
     /**
+     * Output a message in the console stating that the command can not be found and tries to guess what the user
+     * wanted to say. Output a list of available subcommands as well.
+     *
+     * @param string $command Unknown command name trying to be dispatched.
+     * @return void
+     */
+    protected function outputUnknownCommand($command)
```

Comment:
> Shell should be a starting point for userland classes.
Fair enough.

I can put the method in `ConsoleOptionParser` to build the message to be displayed and just have a call to `out()` in `Shell`.